### PR TITLE
feat(declarative_test.go): Add Enhance Skip Reconcile Label Test

### LINKED ISSUE #965: Add a test for the skip-reconciliation for the Manifest Controller
**Description:**

As a followup to this [issue](https://github.com/kyma-project/lifecycle-manager/issues/937), we need to test the functionality of the skip-reconciliation for the Manifest Controller

**ACs:**
- [x] Add an E2E test for the Manifest Controller [here](https://github.com/kyma-project/lifecycle-manager/blob/main/internal/declarative/v2/test/v1/declarative_test.go) to validate that the reconciliation is skipped if the `operator.kyma-project.io/skip-reconciliation` label exists.

Given kyma CR deployed in KCP Cluster
When enabling Template Operator
Then the Sample CR is deployed on the SKR cluster in Ready state

Given Manifest CR with skip label
When deleting Template Operator deployment
Then the deployment should not been recreated.
When deleting Sample CR
Then the Sample CR should not been recreated.

When removing Manifest CR skip label
Then the deployment should be created.
Then the Sample CR should be created.

### PR DESCRIPTION
**Pull request by LeelaChacha**
Closes #965

### CODE REVIEW

**Comment by LeelaChacha** [comment]
@nesmabadr @ruanxin I was unable to resolve the conflict in the last PR because the file was completely moved and the history got too hard to follow

**Review by ruanxin — CHANGES_REQUESTED** [review]
This test should implemented as e2e test, but not an integration test. as mentioned in original AC.

@nesmabadr, can you maybe refactor the AC to the BDD format, so that the expectation is more clear for this task? You can follow this example [issue](https://github.com/kyma-project/lifecycle-manager/issues/1031)

**Comment by ruanxin** [comment]
I added the e2e test scenario in the original [issue](https://github.com/kyma-project/lifecycle-manager/issues/965). @LeelaChacha

File: tests/e2e_test/utils_test.go [hunk 1]
```diff
@@ -148,6 +180,24 @@ func CheckIfNotExists(ctx context.Context, name, namespace, group, version, kind
 	return fmt.Errorf("%w: %s %s/%s should be deleted", errResourceExists, kind, namespace, name)
 }
 
+func DeleteCRWithGVK(ctx context.Context, clnt client.Client, name, namespace, group, version, kind string) error {
```
[the remark this answers is not in the document]
Comment (LeelaChacha) [reply tests/e2e_test/utils_test.go:116]: SetManifestLabels, GetManifestLabels and other Manifest related were moved

[the remark this answers is not in the document]
Comment (ruanxin) [reply tests/e2e_test/utils_test.go:116]: good you aware of this dependencies problem 👍 That's why we already design the [template-operator API](https://github.com/kyma-project/template-operator/issues/102) as a dedicate go module.

Unfortunately, the [PR](https://github.com/kyma-project/lifecycle-manager/pull/982) to introduce template-operator API as dependencies is still on going.

File: tests/e2e_test/manifest_reconciliation_test.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+		It("When the Manifest is labelled to skip reconciliation", func() {
+			labels, err := GetManifestLabels(ctx, kyma.GetName(), kyma.GetNamespace(), module.Name, controlPlaneClient)
+			Expect(err).ToNot(HaveOccurred())
+			labels[declarativev2.SkipReconcileLabel] = "true"
+			err = SetManifestLabels(ctx, kyma.GetName(), kyma.GetNamespace(), module.Name, controlPlaneClient, labels)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("When deleting the SKR Default CR")
+			/*err = SetFinalizer("sample-yaml", "kyma-system", "operator.kyma-project.io",
```
Comment (ruanxin) [inline tests/e2e_test/manifest_reconciliation_test.go:54]: Please remove this comment out code

File: tests/e2e_test/manifest_reconciliation_test.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+			By("Then Module Manager Deployment is not recreated on the SKR cluster")
+			Consistently(CheckIfNotExists).
+				WithContext(ctx).
+				WithArguments("template-operator-controller-manager", "template-operator-system",
+					"apps", "v1", "Deployment", runtimeClient).
+				Should(Succeed())
+		})
+
+		It("When the Manifest skip reconciliation label removed",
+			func() {
```
Comment (ruanxin) [inline tests/e2e_test/manifest_reconciliation_test.go:80]: is it really necessary to make this `func()` here in a separate line? it's regarding the indenting of function body alignment, this whole function body is not aligned with other function which affect the format of the code.

Can you reformate it?

File: tests/e2e_test/utils_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ruanxin) [reply tests/e2e_test/utils_test.go:116]: and you can consider put this new function under `pkg/testutils/modulecr.go`

File: tests/e2e_test/manifest_reconciliation_test.go (hunk 2 above)
Comment (LeelaChacha) [reply tests/e2e_test/manifest_reconciliation_test.go:54]: im not finished with the test. It has a peculiar behaviour in the local testing. I'll let you know when it's ready for review

File: tests/e2e_test/manifest_reconciliation_test.go [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+				WithContext(ctx).
+				WithArguments(runtimeClient, "sample-yaml", "kyma-system", "operator.kyma-project.io",
+					"v1alpha1", "Sample").
+				Should(Succeed())
+			By("Then SKR Module Default CR is not recreated")
+			Consistently(CheckIfExists).
+				WithContext(ctx).
+				WithArguments("sample-yaml", "kyma-system", "operator.kyma-project.io",
+					"v1alpha1", "Sample", runtimeClient).
+				ShouldNot(Succeed())
```
Comment (ruanxin) [inline tests/e2e_test/manifest_reconciliation_test.go:64]: please use `Should(Equal(ErrNotFound))`, `ShouldNot(Succeed())` doesn't mean it's not exists, it may occurred by other error

File: tests/e2e_test/manifest_reconciliation_test.go [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+			By("When deleting the SKR Module Manager Deployment")
+			err = DeleteCRWithGVK(ctx, runtimeClient, "template-operator-controller-manager",
+				"template-operator-system", "apps", "v1", "Deployment")
+			Expect(err).ToNot(HaveOccurred())
+			By("Then Module Manager Deployment is not recreated on the SKR cluster")
+			Consistently(CheckIfExists).
+				WithContext(ctx).
+				WithArguments("template-operator-controller-manager", "template-operator-system",
+					"apps", "v1", "Deployment", runtimeClient).
+				ShouldNot(Succeed())
```
Comment (ruanxin) [inline tests/e2e_test/manifest_reconciliation_test.go:75]: same here

File: tests/e2e_test/manifest_reconciliation_test.go [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+	CleanupKymaAfterAll(kyma)
+
+	Context("Given kyma deployed in KCP", func() {
+		It("When enabling Template Operator", func() {
+			Eventually(EnableModule).
+				WithContext(ctx).
+				WithArguments(runtimeClient, defaultRemoteKymaName, remoteNamespace, module).
+				Should(Succeed())
+			By("Then the Module Operator is deployed on the SKR cluster")
+			Eventually(CheckIfExists).
```
Comment (ruanxin) [inline tests/e2e_test/manifest_reconciliation_test.go:29]: please use `DeploymentIsReady`, it checks if deployment is available.

File: tests/e2e_test/manifest_reconciliation_test.go (hunk 4 above)
Comment (ruanxin) [reply tests/e2e_test/manifest_reconciliation_test.go:64]: for this, you need to merge main branch, this `CheckIfExists` was refactored to call `CRExists` [recently](https://github.com/kyma-project/lifecycle-manager/blob/c710461f7aa929c06f0d892f8dc251d9bda84625/tests/e2e_test/utils_test.go#L131), which made it possible to distinguish the error case.

**Review by ruanxin — APPROVED** [review]
